src/assembly/local/quadrature_table.rs

```diff
@@ -147,6 +147,38 @@ where
             data: self.data,
         }
     }
+
+    /// Replaces the data of the quadrature table with the given data.
+    pub fn with_data<NewData>(self, data: NestedVec<NewData>) -> GeneralQuadratureTable<T, GeometryDim, NewData> {
+        GeneralQuadratureTable {
+            points: self.points,
+            weights: self.weights,
+            data: data,
+        }
```

Comment:
You need to call `Self::from_points_weights_and_data` to ensure that the input is validated!